### ISSUE TITLE
How to view progress on Tensorboard

### ISSUE THREAD

**Issue by SamihaSara** [foolwood/SiamMask#163]
I tried to see  progress on Tensorboard ( logs at SiamMask/experiments/siammasm_base/logs/) with the following command by port forwarding from server to local pc.

ssh -L 16006:127.0.0.1:6006 user@host
Then you run the tensorboard command:
tensorboard --logdir=/path/to/logs
then tried to access the tensorboard in browser under: localhost:16006/

But it did not output any graph or update. If someone has completed this step, please help.

**Comment by tedrostime** [comment]
%tensorboard --logdir [directory name of the log files]

**Comment by SamihaSara** [comment]
@tedrostime Hi, thanks for helping. I have tried this also but got this error--  -bash: fg: %tensorboard: no such job.
Also, I am using 2 GPUs where the paper uses 4. Can you suggest which parameters should be modified for this?

**Comment by tedrostime** [comment]
I think you included the '%' symbol. you shouldn't include the % symbol. 
Just 'tensorflow --logdir [your directory name]' works

**Comment by SamihaSara** [comment]
I tried 'tensorflow --logdir [your directory name]', got this output: 
TensorBoard 1.13.1 at http://devcube4:6007 (Press CTRL+C to quit)
But how to visualize this link? I have tried it on chrome in pc, it shows nothing.
![image](https://user-images.githubusercontent.com/9627710/88495102-39ffe000-cffc-11ea-96cf-b8b4b66c9df5.png)

**Comment by tedrostime** [comment]
the directory name is saved in 'board'. you can check that in tools/train_siammask.py (line # 63)
so the tensorboard will be as:
tensorboard --logdir board/ (something like that)

**Comment by SamihaSara** [comment]
Just tried this. Again nothing in the browser. Can you also tell, how should I open the link -- http://devcube4:6007?

**Comment by mesmertrack** [comment]
> Just tried this. Again nothing in the browser. Can you also tell, how should I open the link -- http://devcube4:6007?

I think copy and paste the link to your browser. 

as a last option, you can try this:
tensorboard --logdir=board/ --host localhost --port 8088
(make sure your path to the logdir is the correct)

**Comment by SamihaSara** [comment]
![image](https://user-images.githubusercontent.com/9627710/88498387-53a62500-d006-11ea-92a3-b318e2cc88c7.png)

**Comment by mesmertrack** [comment]
I think this will work now. copy & paste the link given (http://localhost:8089). 
hope that works

**Comment by SamihaSara** [comment]
![image](https://user-images.githubusercontent.com/9627710/88499375-fe1f4780-d008-11ea-8c0a-1e67eec76657.png)
![image](https://user-images.githubusercontent.com/9627710/88499400-0ecfbd80-d009-11ea-90d2-949a05f8799a.png)

Sorry, I did not work again.

**Comment by mesmertrack** [comment]
> ![image](https://user-images.githubusercontent.com/9627710/88499375-fe1f4780-d008-11ea-8c0a-1e67eec76657.png)
> ![image](https://user-images.githubusercontent.com/9627710/88499400-0ecfbd80-d009-11ea-90d2-949a05f8799a.png)
> 
> Sorry, I did not work again.

in the browser, you are writing 8088, but from the terminal it is in port 8089. can you check that part?

**Comment by SamihaSara** [comment]
Sorry I provided the wrong picture. 
![Screenshot from 2020-07-27 13-04-28](https://user-images.githubusercontent.com/9627710/88499641-cebd0a80-d009-11ea-91cc-58b2ee5d13a0.png)

**Comment by mesmertrack** [comment]
I don't know why but can you try this again:
tensorboard --logdir board/

**Comment by SamihaSara** [comment]
@mesmertrack tried, does not work. Just to confirm, Am I supposed to open the link in the local machine's browser? Is there any way to open this link on the server?

**Comment by mesmertrack** [comment]
oh wow, I have no idea why it is not working. hope someone else helps in this. 
As your question, I think since you are training your dataset in your local machine, opening the link in local browser makes sense. 
Good luck

**Comment by SamihaSara** [comment]
@mesmertrack  no I am training the model on a server. Can you tell me if I can open the link there?

**Comment by mesmertrack** [comment]
I am not quite sure but this link may help. 
https://stackoverflow.com/questions/38464559/how-to-locally-view-tensorboard-of-remote-server/49101487

**Comment by SamihaSara** [comment]
@mesmertrack Thank you so much. One of the solutions from this link worked. Here it is

ssh to your server and find its IP via terminal by running: IP=`hostname -I`
Open the tensorboard server on the host server:
tensorboard --logdir=board --host $IP
Use your browser to and surf to http://$IP:6006

**Comment by mesmertrack** [comment]
great. glad it worked. good luck on your works!

**Comment by SamihaSara** [comment]
@mesmertrack can you please provide some info to debug this by mentioning which parts of the project contain codes for these tasks: Mask representation, ROWs refinement, and Box generation?

**Comment by cnctdadots** [comment]
@SamihaSara : if you check the tools/train_siammask.py, it has train(...) function. you can check the working flow from there.
or to make it simple, start from the main() function.

**Comment by SamihaSara** [comment]
@cnctdadots Can you please tell where and why in the SiamMask architecture, SiamRPN is used? I have read the paper but could not figure out.
